File: csharp/cs prg/WindowsApplicationThread/WindowsApplicationThread/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using System.Threading;

namespace WindowsApplicationThread
{
    public partial class Form1 : Form
    {
        Thread t1, t2;
        
        public Form1()
        {
            InitializeComponent();
        
           t1 = new Thread(new ThreadStart(A));
           t2 = new Thread(new ThreadStart(B));
        }

        private void Form1_Load(object sender, EventArgs e)
        {
  
        }
        int i, j;
        string tname1,tname2;
        
        public void A()
        {
            tname1 = Thread.CurrentThread.Name;
            for (i = 1; i <= 10; i++)
            {
                Console.WriteLine(Thread.CurrentThread.Name + " : A " + i);
               //error//listBox1.Items.Add (Thread.CurrentThread.Name + " : A " + i);
                 AccessConTrolA();
                Thread.Sleep(1000);
            }
        }

        public void B()
        {
           tname2 = Thread.CurrentThread.Name;
            for (j = 1; j <= 20; j++)
            {
                Console.WriteLine(Thread.CurrentThread.Name + " : B " + j.ToString());
                //listBox1.Items.Add (Thread.CurrentThread.Name + " : B " + i.ToString());
                AccessConTrolB();
                Thread.Sleep(500);
            }
        }
        
        void AccessConTrolA()
        {
            if (this.InvokeRequired)
                this.Invoke(new MethodInvoker(AccessConTrolA));
            else
                listBox1.Items.Add(tname1 + " : A " + i);
        }
        
        void AccessConTrolB()
        {
            if (this.InvokeRequired)
                this.Invoke(new MethodInvoker(AccessConTrolB));
            else
                listBox1.Items.Add(tname2 + " : B " + j);
        }
        

        private void button1_Click(object sender, System.EventArgs e)
        {
            t1.Name = "First Thread";
            t2.Name = "Second Thread";

            t2.Priority = ThreadPriority.Highest;

            t1.Start();
            t2.Start();
        }

        private void button3_Click(object sender, System.EventArgs e)
        {
            t1.Suspend();
        }

        private void button4_Click(object sender, System.EventArgs e)
        {
            t2.Suspend();
        }

        private void button5_Click(object sender, System.EventArgs e)
        {
            try
            {
                t1.Resume();
            }
            catch (System.Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void button6_Click(object sender, System.EventArgs e)
        {
            try
            {
                t2.Resume();
            }
            catch (System.Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void button2_Click(object sender, System.EventArgs e)
        {
            t1.Abort();
        }

        private void button7_Click(object sender, System.EventArgs e)
        {
            t2.Abort();
       }
    }
}

File: VB.NET/windows/WindowsThreadApplication/WindowsThreadApplication/Form1.vb
Imports System.Threading
Public Class Form1
    Private t1, t2 As Thread
    Dim i, j As Integer, tname1, tname2 As String

    Sub fun1()
        For i = 1 To 10
            tname1 = Thread.CurrentThread.Name
            AccessControl1()
            Thread.Sleep(1000)
        Next
    End Sub


    Sub AccessControl1()
        If (Me.InvokeRequired) Then
            Me.Invoke(New MethodInvoker(AddressOf accesscontrol1))
        Else
            ListBox1.Items.Add(tname1 & " : " & i)
        End If
    End Sub

    Sub fun2()

        For j = 1 To 20
            tname2 = Thread.CurrentThread.Name
            AccessControl2()
            Thread.Sleep(500)
        Next
    End Sub

    Sub AccessControl2()
        If (Me.InvokeRequired) Then
            Me.Invoke(New MethodInvoker(AddressOf AccessControl2))
        Else
            ListBox1.Items.Add(tname2 & " : " & j)
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        t1 = New Thread(AddressOf fun1)
        t2 = New Thread(AddressOf fun2)

        t1.Name = InputBox("Enter First Thread Name ")
        t2.Name = InputBox("Enter Second Thread Name ")

        t1.Priority = ThreadPriority.BelowNormal

        t1.Start()
        t2.Start()
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        If t1.ThreadState <> ThreadState.Stopped Then
            t1.Abort()
        End If
    End Sub

    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        If t2.ThreadState <> ThreadState.Stopped Then
            t2.Abort()
        End If
    End Sub

    Private Sub Button5_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button5.Click
        t1.Suspend()
    End Sub

    Private Sub Button4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button4.Click
        t2.Suspend()
    End Sub

    Private Sub Button6_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button6.Click
        If t1.ThreadState = ThreadState.Suspended Then
            t1.Resume()
        End If
    End Sub

    Private Sub Button7_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button7.Click
        If t2.ThreadState = ThreadState.Suspended Then
            t2.Resume()
        End If
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

    End Sub
End Class
